synthesizer/src/process/mod.rs

```diff
@@ -469,12 +477,20 @@ mod tests {
         assert_eq!(authorization.len(), 1);
 
         // Execute the request.
-        let (response, execution, _inclusion, _metrics) =
-            process.execute::<CurrentAleo, _>(authorization, rng).unwrap();
+        let mut all_assignments = Vec::with_capacity(1);
+        let (_response, mut execution, _inclusion, _metrics, function_assignments) =
+            process.prepare_function::<CurrentAleo, _>(authorization, rng).unwrap();
+        let function_assignments = Arc::try_unwrap(function_assignments).unwrap_or_default().into_inner(); // TODO: return Err if there is one 
```

Comment:
IIRC an existence of `Arc` clones here would indicate a leak/logic error

Reply:
Yes in a next iteration I'm following the same approach we use for e.g. execution/inclusion, just try_unwrap